rest-api/dao/participant_dao.py

```diff
@@ -117,9 +117,9 @@ def _do_update(self, session, obj, existing_obj):
     # If the provider link changes, update the HPO ID on the participant and its summary.
     if obj.hpoId is None:
       obj.hpoId = existing_obj.hpoId
-    if obj.providerLink != existing_obj.providerLink and obj.providerLink != 'null':
+    if obj.providerLink != existing_obj.providerLink and obj.providerLink is not None:
       new_hpo_id = self._get_hpo_id(obj)
-      if new_hpo_id != existing_obj.hpoId:
+      if new_hpo_id != existing_obj.hpoId and new_hpo_id != 0:
```

Comment:
So... if they *did* specify Organization/UNSET for the provider link, we *should* unpair the participant. I'm not sure we want this zero check here.

Either way, please add a test for that, too.

Reply:
Err. Sorry I take it back. They shouldn't set a provider link in this case. This is fine.